mp/src/game/server/momentum/mom_system_saveloc.cpp

```diff
@@ -436,13 +436,23 @@ SavedLocation_t* CSaveLocSystem::CreateSaveloc(int components /*= SAVELOC_ALL*/)
 
 void CSaveLocSystem::CreateAndSaveLocation()
 {
-    SavedLocation_t *created = CreateSaveloc();
-    if (created)
+    const auto pPlayer = CMomentumPlayer::GetLocalPlayer();
+    if (!pPlayer)
+        return;
+
+    const auto pEntData = pPlayer->GetRunEntData();
+    if (!pEntData)
+        return;
+
+    if (pEntData->m_bIsInZone && pEntData->m_iCurrentZone == 1 && pPlayer->CreateStartMark())
```

Comment:
You can get rid of the first two checks as now, create start mark returning false means they also weren't in the start zone. This also means ridding the entdata fetch as well.

Reply:
oh! 🧠